packages/react/toast/src/Toast.tsx

```diff
@@ -173,36 +173,47 @@ const ToastViewport = React.forwardRef<ToastViewportElement, ToastViewportProps>
     React.useEffect(() => {
       const wrapper = wrapperRef.current;
       const viewport = ref.current;
-      if (wrapper && viewport) {
+      if (hasToasts && wrapper && viewport) {
         const handlePause = () => {
-          const pauseEvent = new CustomEvent(VIEWPORT_PAUSE);
-          viewport.dispatchEvent(pauseEvent);
-          context.isClosePausedRef.current = true;
+          if (!context.isClosePausedRef.current) {
+            const pauseEvent = new CustomEvent(VIEWPORT_PAUSE);
+            viewport.dispatchEvent(pauseEvent);
+            context.isClosePausedRef.current = true;
+          }
         };
 
         const handleResume = () => {
-          const resumeEvent = new CustomEvent(VIEWPORT_RESUME);
-          viewport.dispatchEvent(resumeEvent);
-          context.isClosePausedRef.current = false;
+          if (context.isClosePausedRef.current) {
+            const resumeEvent = new CustomEvent(VIEWPORT_RESUME);
+            viewport.dispatchEvent(resumeEvent);
+            context.isClosePausedRef.current = false;
+          }
+        };
+
+        const handleFocusOutResume = (event: FocusEvent) => {
+          // Only resume when focus moves out of the wrapper
+          if (!wrapper.contains(event.relatedTarget as HTMLElement)) {
```

Comment:
We still need to handle for this as `focusout / focusin` will fire while tabbing within wrapper and we don't want to flip the flag until we're back outside.